infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/ElementListing.kt

```diff
@@ -83,11 +86,15 @@ fun toElementListing(
     getSwitchName: (IntId<TrackLayoutSwitch>) -> SwitchName,
 ): List<ElementListing> {
     val linkedElementIds = collectLinkedElements(layoutAlignment.segments, context, startAddress, endAddress)
-    val linkedAlignmentIds = linkedElementIds.mapNotNull { (_, id) -> id?.let(::getAlignmentId) }.distinct()
+    val lengthOfSegmentsConnectedToSameElement = linkedElementIds.groupBy { it.second }.map {
+        it.key to it.value.map { (segment, _) -> segment.length }.sum()
+    }
```

Comment:
Yksi geometriaelementti voi liittyä useaan segmenttiin -> groupataan segmentit sen mukaan mihin segmenttiin ne kuuluvat ja summataan ka. segmenttien pituudet yhteen. Näitä verrataan myöhemmin näihin liittyvien elementtien pituuksiin.